colossalai/tensor/colo_tensor.py

```diff
@@ -142,89 +52,51 @@ def is_model_data(self) -> bool:
 
     @classmethod
     def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+
+        if not all(issubclass(cls, t) for t in types):
+            return NotImplemented
         global _COLOSSAL_OPS
         if func in _COLOSSAL_OPS:
-            for arg in args:
-                if isinstance(arg, ColoTensor):
-                    return _COLOSSAL_OPS[func](types, args, kwargs, None)
-
-            for kwarg in kwargs.values():
-                if isinstance(kwarg, ColoTensor):
-                    return _COLOSSAL_OPS[func](types, args, kwargs, None)
-        else:
-            # If we have not hijact the function, convert the ColoTensors in args and kwargs to torch tensors.
-            args = [arg.torch_tensor() if isinstance(arg, ColoTensor) else arg for arg in args]
-            if kwargs is None:
-                kwargs = {}
-
-            kwargs = {k: v.torch_tensor() if isinstance(v, ColoTensor) else v for k, v in kwargs.items()}
-            return cls._filter_outputs_with_colo(func(*args, **kwargs))
-
-    def backward(self, gradient: Optional[torch.Tensor] = None, retain_graph: bool = False):
-        self._torch_tensor.backward(gradient=gradient, retain_graph=retain_graph)
-
-    def __add__(self, o) -> "ColoTensor":
-        if isinstance(o, ColoTensor):
-            return ColoTensor.init_from_torch_tensor(self.torch_tensor() + o.torch_tensor())
-        elif isinstance(o, (torch.Tensor, int, float)):
-            return ColoTensor.init_from_torch_tensor(self.torch_tensor() + o)
-        else:
-            raise TypeError(f'{type(o)} is not supported in ColoTensor __add__')
-
-    __radd__ = __add__
-
-    def __truediv__(self, o) -> "ColoTensor":
-        return ColoTensor.init_from_torch_tensor(self.torch_tensor() / o)
-
-    def __getattr__(self, name):
+            func = _COLOSSAL_OPS[func]
 
-        def replace_tensor_with_colo(func):
+        with torch._C.DisableTorchFunction():
+            ret = func(*args, **kwargs)
+            if func in get_default_nowrap_functions():
+                return ret
+            else:
+                return _convert_output(ret)
```

Comment:
I remember you use `super.__torch__function__` here yesterday?
Why changing?

Reply:
I just copy the content of `torch.Tensor.__torch_function__` here, since `torch.Tensor.__torch_function__` convert the output tensor to current class. However, for ColoParameter, no matther what op is applied, it should return a ColoTensor rather than ColoParameter.